iroh-cli/tests/cli.rs

```diff
@@ -142,74 +142,98 @@ fn cli_provide_tree_resume() -> Result<()> {
-            } else {
-                MakePartialResult::Retain
-            }
-        })?;
-        let get_output = run_get_cmd(&get_iroh_data_dir, &ticket, Some(tgt.clone()))?;
-        let matches = explicit_matches(match_get_stderr(get_output.stderr)?);
-        assert_eq!(matches, vec!["98.04 KiB"]);
-        compare_files(&src, &tgt)?;
-        std::fs::remove_dir_all(&tgt)?;
+    // setup the data dir for the iroh instances that will get the blobs
+    let src_iroh_data_dir = tmp.join("src_iroh_data_dir");
+    copy_blob_dirs(&src_iroh_data_dir_pre, &src_iroh_data_dir)?;
+    // first tests
+    let empty_data_dir = tmp.join("get_iroh_data_dir_01");
+    // second test
+    let full_data_dir = tmp.join("get_iroh_data_dir_02");
+    copy_blob_dirs(&src_iroh_data_dir, &full_data_dir)?;
+    // third test
+    let missing_blobs_data_dir = tmp.join("get_iroh_data_dir_03");
+    copy_blob_dirs(&src_iroh_data_dir, &missing_blobs_data_dir)?;
+    make_partial(&missing_blobs_data_dir, |_hash, size| {
+        if size == 100000 {
+            MakePartialResult::Remove
+        } else {
+            MakePartialResult::Retain
+        }
+    })?;
+    // fourth test
+    let partial_data_dir = tmp.join("get_iroh_data_dir_04");
+    copy_blob_dirs(&src_iroh_data_dir, &partial_data_dir)?;
+    make_partial(&partial_data_dir, |_hash, size| {
+        if size == 100000 {
+            MakePartialResult::Truncate(1024 * 32)
+        } else {
+            MakePartialResult::Retain
+        }
+    })?;
+
+    struct TestCase {
+        name: &'static str,
+        get_folder: PathBuf,
+        transfer_size: &'static str,
     }
 
-    // fourth test - partial work dir - truncate some large files
-    {
-        println!("fourth test - partial work dir - truncate some large files");
-        let get_iroh_data_dir = tmp.join("get_iroh_data_dir_04");
-        copy_blob_dirs(&src_iroh_data_dir, &get_iroh_data_dir)?;
-        make_partial(&get_iroh_data_dir, |_hash, size| {
-            if size == 100000 {
-                MakePartialResult::Truncate(1024 * 32)
-            } else {
-                MakePartialResult::Retain
-            }
-        })?;
-        let get_output = run_get_cmd(&get_iroh_data_dir, &ticket, Some(tgt.clone()))?;
+    let test_cases = [
+        TestCase {
+            name: "empty folder produces full transfer",
+            get_folder: empty_data_dir,
+            transfer_size: "112.89 KiB",
+        },
+        TestCase {
+            name: "full folder needs no transfer",
+            get_folder: full_data_dir,
+            transfer_size: "0 B",
+        },
+        TestCase {
+            name: "missing blobs produces partial transfer",
+            get_folder: missing_blobs_data_dir,
+            transfer_size: "98.04 KiB",
+        },
+        TestCase {
+            name: "partial blobs produces partial transfer",
+            get_folder: partial_data_dir,
+            transfer_size: "65.98 KiB",
+        },
+    ];
```

Comment:
Do you have thoughts on using a `struct TestCase` vs. writing a small `fn test_case(name: &'static str, get_folder: PathBuf, transfer_size: &'static str)` that is called four times?

Could also make it not a `fn`, but a closure instead. That way it can capture references from the environment, thus not needing extra arguments beyond the three.

Reply:
I would also prefer something less verbose, but I would much rather have a struct that very explicitly lets us know what is that we are testing, than go back to the previous state where, at least imo, it was not clear what the test is intended to assert. I'll give it a try to maybe a fn instead of a loop, and yeah if it's indeed equally clear but less verbose let's go with it

Reply:
ok, this required more refactoring and renaming to make cargofmt fit each test on one line and actually reduce line count, but at the end it's indeed less verbose :)